crates/search/src/generation/post_apply.rs

```diff
@@ -261,11 +261,11 @@ impl CandidateGeneration for PostApplyCandidateGeneration {
             .collect::<Vec<_>>();
 
         let need_post_apply = !unapplied_filters.is_empty() || !unapplied_projection.is_empty();
-        let underlying_limit = if need_post_apply {
+        let underlying_limit = if unapplied_filters.is_empty() {
```

Comment:
This is a separate improvement I found. We only need a higher limit iff there are unapplied filters.